test/user_test.rb

```diff
@@ -0,0 +1,59 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require './lib/user'
+
+class UserTest < Minitest::Test
+
+  def test_it_exists
+    sal = User.new("Sal")
+    assert_instance_of User, sal
+  end
+
+  def test_it_has_a_name
+    sal = User.new("Sal")
+    assert_equal "Sal", sal.name
+  end
+
+  def test_user_starts_with_no_jokes
+    sal = User.new("Sal")
+    assert_equal [], sal.jokes
+  end
+
+  def test_user_can_learn_jokes
+    sal = User.new("Sal")
+    joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+    sal.learn(joke_1)
+    assert_instance_of Joke, sal.jokes[0]
+    sal.learn(joke_2)
+    assert_instance_of Joke, sal.jokes[1]
+  end
+
+  def test_user_can_tell_joke
+    sal = User.new("Sal")
+    ali = User.new("Ali")
+    joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+    sal.learn(joke_1)
+    sal.learn(joke_2)
+    sal.tell(ali, joke_1)
+    sal.tell(ali, joke_2)
+    assert_equal 2, ali.jokes.length
+    assert_equal "How do you keep a lion from charging?", ali.jokes[1].setup
```

Comment:
You could combine and shorten these two assertions like this: `assert_equal [joke_1, joke_2], ali.jokes`